frontend/packages/console-dynamic-plugin-sdk/src/webpack/ExtensionValidatorPlugin.ts

```diff
@@ -0,0 +1,41 @@
+import { dirname } from 'path';
+import { WebpackPluginInstance, Compiler, WebpackError } from 'webpack';
+import { getExtensionsFilePath } from '@console/plugin-sdk/src/codegen/active-plugins';
+import { PluginPackage } from '@console/plugin-sdk/src/codegen/plugin-resolver';
+import { ConsolePluginPackageJSON } from '../build-types';
+import { ConsoleExtensionsJSON } from '../schema/console-extensions';
+import { parseJSONC } from '../utils/jsonc';
+import { ExtensionValidator } from '../validation/ExtensionValidator';
+
+export type ExtensionValidatorPluginOptions = {
+  /** List of plugin packages to validate */
+  pluginPackages: PluginPackage[] | ConsolePluginPackageJSON[];
+};
+
+/**
+ * Validate the integrity of the exposed modules and codeRefs for static
+ * plugin packages.
+ */
+export class ExtensionValidatorPlugin implements WebpackPluginInstance {
+  constructor(private readonly options: ExtensionValidatorPluginOptions) {}
```

Comment:
Good webpack plugins generally provide sensible defaults for all of their options, so this could be:
```ts
  constructor(private readonly options: ExtensionValidatorPluginOptions = {}) {}
```

But in this particular case, one may argue that `pluginPackages` is essential and should not be empty, so I'd suggest to include a basic sanity check within the constructor:

```ts
constructor(private readonly options: ExtensionValidatorPluginOptions) {
  if (options.pluginPackages.length === 0) {
    throw new Error('List of plugin packages to validate must not be empty');
  }
}
```